pkg/querier/queryrange/codec_test.go

```diff
@@ -202,6 +202,57 @@ func Test_codec_EncodeDecodeRequest(t *testing.T) {
 			Step:        30 * 1e3, // step is expected in ms; default is 0 or no step
 			AggregateBy: "series",
 		}, false},
+		{"detected_fields", func() (*http.Request, error) {
+			return DefaultCodec.EncodeRequest(ctx, &DetectedFieldsRequest{
```

Comment:
I'm a bit confused as to why we use a wrapped queryrange.DetectedFieldsRequest and not the raw logproto.DetectedFieldsRequest here - is this the right thing to do?

Reply:
the query-frontend is messy for sure I think it's fine.